flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/data/StructRowData.java

```diff
@@ -0,0 +1,300 @@
+      throw new IllegalStateException(
+          "Unknown type for long field. Type name: " + longVal.getClass().getName());
+    }
+  }
+
+  @Override
+  public float getFloat(int pos) {
+    return struct.get(pos, Float.class);
+  }
+
+  @Override
+  public double getDouble(int pos) {
+    return struct.get(pos, Double.class);
+  }
+
+  @Override
+  public StringData getString(int pos) {
+    return isNullAt(pos) ? null : getStringDataInternal(pos);
+  }
+
+  private StringData getStringDataInternal(int pos) {
+    CharSequence seq = struct.get(pos, CharSequence.class);
+    return StringData.fromString(seq.toString());
+  }
+
+  @Override
+  public DecimalData getDecimal(int pos, int precision, int scale) {
+    return isNullAt(pos)
+        ? null
+        : DecimalData.fromBigDecimal(getDecimalInternal(pos), precision, scale);
+  }
+
+  private BigDecimal getDecimalInternal(int pos) {
+    return struct.get(pos, BigDecimal.class);
+  }
+
+  @Override
+  public TimestampData getTimestamp(int pos, int precision) {
+    long timeLong = getLong(pos);
+    return TimestampData.fromEpochMillis(timeLong / 1000, (int) (timeLong % 1000) * 1000);
+  }
+
+  @Override
+  public <T> RawValueData<T> getRawValue(int pos) {
+    throw new UnsupportedOperationException("Not supported yet.");
+  }
+
+  @Override
+  public byte[] getBinary(int pos) {
+    return isNullAt(pos) ? null : getBinaryInternal(pos);
+  }
+
+  private byte[] getBinaryInternal(int pos) {
+    Object bytes = struct.get(pos, Object.class);
+
+    // should only be either ByteBuffer or byte[]
+    if (bytes instanceof ByteBuffer) {
+      return ByteBuffers.toByteArray((ByteBuffer) bytes);
+    } else if (bytes instanceof byte[]) {
+      return (byte[]) bytes;
+    } else if (bytes instanceof UUID) {
+      UUID uuid = (UUID) bytes;
+      ByteBuffer bb = ByteBuffer.allocate(16);
+      bb.putLong(uuid.getMostSignificantBits());
+      bb.putLong(uuid.getLeastSignificantBits());
+      return bb.array();
+    } else {
+      throw new IllegalStateException(
+          "Unknown type for binary field. Type name: " + bytes.getClass().getName());
+    }
+  }
+
+  @Override
+  public ArrayData getArray(int pos) {
+    return isNullAt(pos)
+        ? null
+        : (ArrayData)
+            coverValue(type.fields().get(pos).type().asListType(), struct.get(pos, List.class));
+  }
+
+  @Override
+  public MapData getMap(int pos) {
+    return isNullAt(pos)
+        ? null
+        : (MapData)
+            coverValue(type.fields().get(pos).type().asMapType(), struct.get(pos, Map.class));
+  }
+
+  @Override
+  public RowData getRow(int pos, int numFields) {
+    return isNullAt(pos) ? null : getStructRowData(pos, numFields);
+  }
+
+  private StructRowData getStructRowData(int pos, int numFields) {
+    return new StructRowData(
+        type.fields().get(pos).type().asStructType(), struct.get(pos, StructLike.class));
+  }
+
+  private <T> Object coverValue(Type elementType, Object value) {
```

Comment:
should be `convertValue`. `cover` is a different word and meaning

Reply:
thx, done.